lib/descriptor/type/addr.js

```diff
@@ -0,0 +1,65 @@
+/*!
+ * addr.js - address descriptor object for bcoin
+ * Copyright (c) 2023, the bcoin developers (MIT License).
+ * https://github.com/bcoin-org/bcoin
+ */
+
+'use strict';
+
+const Descriptor = require('../descriptor');
+const assert = require('bsert');
+const Address = require('../../primitives/address');
+const Network = require('../../protocol/network');
+
+/**
+ * AddressDescriptor
+ * Parsed address descriptor object
+ * @property {String} name
+ * @property {Address} address
+ * @property {Network} network
+ * @extends Descriptor
+ */
+
+class AddressDescriptor extends Descriptor {
+  constructor(options) {
+    super();
+    this.name = 'addr';
```

Comment:
use `ADDR` from`descriptorTypes` defined in `lib/descriptor/common.js`

Same comment for all files in `lib/descriptor/type`